packages/debug/src/browser/debug-session-label-provider.ts

```diff
@@ -0,0 +1,46 @@
+// *****************************************************************************
+// Copyright (C) 2025 and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import URI from '@theia/core/lib/common/uri';
+import { inject, injectable } from '@theia/core/shared/inversify';
+import { WorkspaceService } from '@theia/workspace/lib/browser';
+import { type DebugSessionOptions } from './debug-session-options';
+
+/**
+ * Provides a label for the debug session without the need to create the session.
+ * Debug session labels are used to check if sessions are the "same".
+ */
+@injectable()
+export class DebugSessionLabelProvider {
```

Comment:
We should call this `DebugSessionConfigurationLabelProvider` to better express its functionality. I noticed that each `DebugSession` actually already has a `LabelProvider` attached with it, so this could be misleading overall.

Reply:
https://github.com/eclipse-theia/theia/pull/14862/commits/142cd4feda368ea7e647736d87ee808f43636119